src/minichain/genesis.py

```
@@ -0,0 +1,81 @@
+        difficulty_target=config.difficulty_target,
+        nonce=0,
+        block_height=0,
+    )
+    return Block(header=header, transactions=[])
+
+
+def apply_genesis_block(state: State, block: Block, config: GenesisConfig) -> None:
+    """Apply genesis allocations to an empty state."""
+    config.validate()
+    if state.accounts:
+        raise ValueError("Genesis can only be applied to an empty state")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)